src/util.cpp

```diff
@@ -160,9 +159,8 @@ class CInit
         RAND_cleanup();
         // Shutdown OpenSSL library multithreading support
         CRYPTO_set_locking_callback(NULL);
-        for (int i = 0; i < CRYPTO_num_locks(); i++)
-            delete ppmutexOpenSSL[i];
-        OPENSSL_free(ppmutexOpenSSL);
```

Comment:
AFAIK it was slightly wrong to free the ppmutexOpenSSL via `OPENSSL_free`. IMO only objects alloced by `OPENSSL_malloc()` should use `OPENSSL_free()`. This change makes sense.